src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -301,5 +304,31 @@ public async Task<ApplicationsViewModel> GetApplications(ApplicationsRequest req
                 })
             };
         }
+
+        public async Task<ApplicationViewModel> GetApplicationForAsync(ApplicationRequest request, CancellationToken cancellationToken = default)
+        {
+            var result =
+                await _pledgeService.GetApplication(request.AccountId, request.PledgeId, request.ApplicationId, cancellationToken);
+
+            if (result != null)
+            {
+                return new ApplicationViewModel(result.Sector, result.AllSectors, result.PledgeSectors, result.PledgeJobRoles, result.PledgeLevels, result.PledgeLocations,
```

Comment:
We should refactor this to not use a parameterised constructor and just set the all the properties here. There are too many parameters for that constructor, and not all properties are set that way anyway.

Reply:
ditched the constructor and moved the code to calculate the match percentage into a method called SetupModel. Happy to change the name if needed. Knowing what's coming in TM-48 I'm thinking this will move to a factory to do the initial list setups and calculations returning the model so the other properties can be populated as it's this functionality that's required in the view and ensures everything is ready wihtout having to call a method to iterate the collection.